File: word2json/models/timeline.py
# %%
from typing import Optional
from word2json.models.base import *


class Timeline(BaseDocument):
    body: Optional[list[Body]]
    category: Optional[str]
    date: Optional[str]
    time: Optional[str]
    order: Optional[int | float]
    tags: Optional[list[str]]
    title: Optional[str]
    source: Optional[str]
    people: Optional[list[str]]
    imageUrls: Optional[list[ImageUrl]]
    articleUrls: Optional[list[ArticleUrl]]

File: json2word/models/timeline.ts
import { Body, BaseDocument, ArticleUrl, ImageUrl } from "./base";

export interface Timeline extends BaseDocument {
	title?: string;
	body?: Body[];
	category?: string;
	date?: string;
	time?: string;
	order?: number;
	tags?: string[];
	source?: string;
	people?: string[];
	articleUrls?: ArticleUrl[];
	imageUrls?: ImageUrl[];
}
